src/scripts/services/user-service.ts

```diff
@@ -109,34 +110,54 @@ export class OktaUserService {
   };
 
   // eslint-disable-next-line functional/functional-parameters
-  readonly listUsers = (): TE.TaskEither<string, readonly User[]> => {
+  readonly listUsers = (
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    maybeGroup: O.Option<Group> = O.none
+  ): TE.TaskEither<string, readonly User[]> => {
     // We need to populate users with all of the client data so it can be
     // returned. Okta's listUsers() function returns a custom collection that
     // does not allow for any form of mapping, so array mutation is needed.
 
-    return TE.tryCatch(
-      // eslint-disable-next-line functional/functional-parameters
-      () => {
-        // eslint-disable-next-line functional/prefer-readonly-type
-        const users: User[] = [];
+    // eslint-disable-next-line functional/prefer-readonly-type
+    const users: User[] = [];
 
-        return (
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const groupOrClient: TE.TaskEither<string, okta.Client | okta.Group> =
+      O.fold(
+        // eslint-disable-next-line functional/functional-parameters, functional/no-this-expression
+        (): TE.TaskEither<string, okta.Client | okta.Group> =>
           // eslint-disable-next-line functional/no-this-expression
-          this.client
-            .listUsers()
-            // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
-            .each((oktaUser) => {
-              // eslint-disable-next-line functional/immutable-data
-              return users.push(oktaUserAsUser(oktaUser));
-            })
-            // eslint-disable-next-line functional/functional-parameters
-            .then(() => {
-              return users;
-            })
-        );
-      },
-      (error: unknown) =>
-        `Failed to list users because of [${JSON.stringify(error)}].`
+          TE.right(this.client),
+        // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+        (group: Group) =>
+          TE.tryCatch(
+            // eslint-disable-next-line functional/functional-parameters, functional/no-this-expression
+            () => this.client.getGroup(group.id),
+            (error: unknown) => `Fail ${JSON.stringify(error)}`
+          )
+      )(maybeGroup);
+
+    return (
```

Comment:
It's so that the maybeGroup parameter is applied to the fold

Reply:
🤔 do you mean so that the `groupOrClient` parameter is applied to the `chain` ?

Reply:
Yes, well the groupOrClient parameter is defined as the end result of the fold, so that it is a TE of either a client or a group (because convineintly both classes have a `listUsers` method). So the goal is to create a fold that converts a potential group into either an actual group (if the group exists) or a client (if no group was given), and then apply the maybeGroup to that fold to get a definite TE value.

Reply:
It looks like I can remove the parens that used to start on line 140 and end on line 161 without issue. I don't understand how they are related to the fold.

Reply:
It seems that was a holdover from previous code or prettier/prettier fixes